tests/IotServiceTest.cpp

```diff
@@ -12,6 +12,11 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
+
+#ifdef _WIN32
```

Comment:
Is this compiler or windows related? My guess is you want to check for msvc not windows....

Reply:
It was actually compiler related, so ``_MSC_VER`` would have been the proper define to check.

However I started thinking that ``snprintf`` also satisfies the msvc compiler, which would make the sample identical for all platforms.